art/attacks/poisoning/perturbations/audio_perturbations.py

```diff
@@ -16,136 +16,110 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 """
-Adversarial perturbations designed to work for images.
+Adversarial perturbations designed to work for audio.
+Uses classes, rather than pure functions as in image_perturbations.py,
+because loading the audio trigger from disk (librosa.load()) is very slow
+and should be done only once.
 """
 import numpy as np
 import librosa
 
 
-def insert_tone_trigger(
-    x: np.ndarray,
-    sampling_rate: int = 16000,
-    frequency: int = 440,
-    duration: float = 0.1,
-    random: bool = False,
-    shift: int = 0,
-    scale: float = 0.1,
-) -> np.ndarray:
+class CacheTrigger:
     """
-    Adds a 'tone' with a given frequency to audio example. Works for a single example or a batch of examples.
+    Adds an audio backdoor trigger to a set of audio examples. Works for a single example or a batch of examples.
 
-    :param x: N x L matrix or length L array, where N is number of examples, L is the length in number of samples.
-              X is in range [-1,1].
-    :param sampling_rate: Positive integer denoting the sampling rate for x.
-    :param frequency: Frequency of the tone to be added.
-    :param duration: Duration of the tone to be added.
+    :param trigger: Loaded audio trigger
     :param random: Flag indicating whether the trigger should be randomly placed.
     :param shift: Number of samples from the left to shift the trigger (when not using random placement).
     :param scale: Scaling factor for mixing the trigger.
     :return: Backdoored audio.
```

Comment:
Return for a class reads a bit awkward. It may be good to remove the return description, since it is explained for the `insert` method.